tests/integration/devfile/cmd_devfile_deploy_test.go

```diff
@@ -62,17 +66,32 @@ var _ = Describe("odo devfile deploy command tests", func() {
 			err := helper.ReplaceDevfileField("devfile.yaml", "dockerfile", "https://google.com")
 			Expect(err).To(BeNil())
 
+			imageTag = fmt.Sprintf("image-registry.openshift-image-registry.svc:5000/%s/my-nodejs:1.0", namespace)
 			cmdOutput := helper.CmdShouldFail("odo", "deploy", "--tag", imageTag, "--devfile", "devfile.yaml")
 			Expect(cmdOutput).To(ContainSubstring("does not point to a valid Dockerfile"))
 		})
 	})
 
+	// This test depends on the nodejs stack to no have a alpha.build-dockerfile field.
+	// This may not be the case in the future when the stack gets updated.
 	Context("Verify error when no Dockerfile exists in project and no 'dockerfile' specified in devfile", func() {
 		It("Should error out with 'dockerfile required for build.'", func() {
 			helper.CmdShouldPass("odo", "create", "nodejs", "--project", namespace, cmpName)
 			helper.CmdShouldPass("odo", "url", "create", "--port", "3000")
+			imageTag = fmt.Sprintf("image-registry.openshift-image-registry.svc:5000/%s/my-nodejs:1.0", namespace)
 			cmdOutput := helper.CmdShouldFail("odo", "deploy", "--tag", imageTag, "--devfile", "devfile.yaml")
```

Comment:
There is still a --devfile being passed here!

Reply:
yup steve's pr fixed that